### DISCUSSION TITLE
Links to headings lost upon export

### DISCUSSION THREAD

**Discussion by Mike221223** [zim-desktop-wiki/zim-desktop-wiki#2264] [Q&A] [answered]
(I'm using Zim 0.75.1 on Linux Mint 20.3)

It appears that links to headings (described [here](https://zim-wiki.org/manual/Help/Links.html#link-to-a-heading-or-object)) are not present in the exported html files.

Here's what happens:

I insert a link to a heading on a page with `#` – for example, `2.1_перевод:SourcePaperKodishB11Ru#источники` – and within Zim it works as described on the website help page – it jumps to the heading `источники`. When I hover my mouse cursor or set the caret inside the text with the link attached to it, it shows the address at the bottom of the page.

However, when I export the page as an html file the link no longer leads to the heading. It appears to simply reload the current page. The link itself changes. Here's a comparison:

Inserted link: `2.1_перевод:SourcePaperKodishB11Ru#источники`
Exported link (shown upon hovering the cursor over it): `SourcePaperKodishB11Ru.html#2.1_перевод:SourcePaperKodishB11Ru`

The exported link doesn't contain the heading name (`источники`) at all. Clicking the link seems to reload the page, or do something else, I don't know what – it certainly doesn't jump to the intended heading.

Yet if I add `#источники` to the address bar – like this: `SourcePaperKodishB11Ru.html#источники` – it takes me to the heading, so I believe this is how it's supposed to work.

I think it actually worked once on an earlier version but only specifically on Google Chrome. I've checked it on Chrome again, and I have the same problem (on the earlier versions of Zim as well). Perhaps it only worked as intended on a specific version of Chrome, which I deleted and now can't check because I don't know which version it was.

Am I doing something wrong?
Is it my configuration?

**Comment by Mike221223** [comment]
I checked the exported html file and found that the line representing the link looks like this:

`<a href="#2.1 перевод:SourcePaperKodishB11Ru" title="#источники" class="page">#источники</a>`

I then replaced the `#2.1 перевод:SourcePaperKodishB11Ru` part with just `#источники`, and again it worked.

Does Zim export the heading links incorrectly?

It's not a problem as long as I only need to export a couple of pages with a few heading links, – I can fix them post-export (even though I have no idea if I'm doing it right) – but if I want to keep a whole base of pages with tables of contents, this means hours of additional work.

**Comment by introt** [comment]
Thanks for reporting this, I've opened ticket #2265

**Reply by introt** [reply, marked as answer]
The fix has been merged
